pkg/antctl/raw/packetcapture/command.go

```
@@ -263,24 +259,63 @@ func parseFlow(options *packetCaptureOptions) (*v1alpha1.Packet, error) {
 		}
 	}
 	if r, ok := fields["tcp_src"]; ok {
+		srcPort, err := strconv.ParseInt(r, 10, 32)
+		if err != nil {
+			return nil, err
+		}
 		pkt.TransportHeader.TCP = new(v1alpha1.TCPHeader)
-		pkt.TransportHeader.TCP.SrcPort = ptr.To(int32(r))
+		pkt.TransportHeader.TCP.SrcPort = ptr.To(int32(srcPort))
 	}
 	if r, ok := fields["tcp_dst"]; ok {
+		dstPort, err := strconv.ParseInt(r, 10, 32)
+		if err != nil {
+			return nil, err
+		}
+		if pkt.TransportHeader.TCP == nil {
+			pkt.TransportHeader.TCP = new(v1alpha1.TCPHeader)
+		}
+		pkt.TransportHeader.TCP.DstPort = ptr.To(int32(dstPort))
+	}
+	if r, ok := fields["tcp_flags"]; ok {
+		flag_matcher := strings.Split(r, "/")
+		if len(flag_matcher) != 2 {
+			return nil, fmt.Errorf("invalid tcp_flags format, expected a/b: %s", r)
+		}
```

why not use a reasonable default when the mask is omitted, just like for the API?

---

we can default it to value

---

updated